### PR TITLE
Add request signing

### PR DESCRIPTION
**Pull request by peeech**
This PR adds:
- X-Holo-Admin-Signature header and signatureto all HTTP calls
- X-Holo-Admin-Signature query string and signature to HTTP GET UPGRADE calls
- x-hp-admin-cas header to HTTP PUT /config calls

### CODE REVIEW

File: src/data-interfaces/HposInterface.js [hunk 1]
```diff
@@ -25,14 +17,17 @@ const axiosConfig = {
...
 }
 
-export function hposCall ({ method = 'get', path, authToken, apiVersion = 'v1', headers: userHeaders = {} }) {
+export function hposCall ({ method = 'get', path, apiVersion = 'v1', headers: userHeaders = {} }) {
   if (MOCK_HPOS_CONNECTION) {
     return mockCallHpos(method, apiVersion, path)
   } else {
     return async params => {
-      const fullPath = process.env.REACT_APP_HPOS_URL + '/' + apiVersion + '/' + path
+      // For Robbie: this url should be probably constructed from window.location, right?
```
Comment (robbiecarlton) [inline src/data-interfaces/HposInterface.js]: having it be an environment variable gives us more control in developing. Currently
`'https://' + window.location.hostname + '/' + apiVersion + '/' + path`. This change will work once we have HPOS as part of our dev environment.

Comment (peeech) [reply src/data-interfaces/HposInterface.js]: 👍

**Comment by robbiecarlton** [comment]
@peeech ok, cleaned this branch up. As far as I'm concerned it's ready to merge. Tagging you here in case you want to take a last look.

**Comment by peeech** [comment]
Great, I already did. For me ready to merge. Trying to make it work on actual HPOS now :-)

**Pull request merged**